R/FilterState.R

```diff
@@ -51,38 +51,38 @@ FilterState <- R6::R6Class( # nolint
     #'   name of the variable
     #' @param varlabel (`character(1)`)\cr
     #'   label of the variable (optional).
-    #' @param input_dataname (`name` or `call`)\cr
+    #' @param dataname (`name` or `call`)\cr
```

Comment:
I assumed you were referring to making `dataname` argument as character class.

If we do this, we need to make sure that `$get_dataname()` is behaving properly.

https://github.com/insightsengineering/teal.slice/blob/aab1255ec1cbb6442f7d3cef1271eed873aec0a7/R/FilterState.R#L180-L186

If we treat `dataname` as character, then it will have backslash `\` when calling `$get_dataname`
```r
# default behavior
r$> filter_state$get_dataname()
[1] "data"

r$> deparse1(as.name("data"))
[1] "data"

# if we treat dataname as character
r$> deparse1("data")
[1] "\"data\""


```

Reply:
Yes, the `deparse` in the method should be removed.

Note the slash is just an esacape character because you get a string containing quote marks.